### ISSUE TITLE
Unexpected error with int called on non-numeric literal

### ISSUE THREAD

**Issue by yuhan0** [borkdude/clj-kondo#588]
**version**

clj-kondo v2019.11.03
**platform**
macOS 10.14
**problem**

clj-kondo throws an exception when the `int` function is called on a non-numeric literal.

**repro**
```bash
$ clj-kondo --lint - <<< '(int "")'
```
The same error happens with `(int :abc)`, `(int [])`, `(int {})` etc. but not with the related fns `float`, `long`, `short`, `byte`.

**expected behavior**

Return a type error / warning

**actual behavior**
```
Unexpected error. Please report an issue.
java.lang.ClassCastException: clojure.lang.PersistentHashSet cannot be cast to clojure.lang.Named
	at clojure.core$namespace.invokeStatic(core.clj:1597)
	at clj_kondo.impl.types$nilable_QMARK_.invokeStatic(types.clj:59)
	at clj_kondo.impl.types$label.invokeStatic(types.clj:101)
	at clj_kondo.impl.types$emit_non_match_BANG_.invokeStatic(types.clj:271)
	at clj_kondo.impl.types$lint_arg_types.invokeStatic(types.clj:373)
	at clj_kondo.impl.linters$lint_arg_types_BANG_.invokeStatic(linters.clj:170)
	at clj_kondo.impl.linters$lint_var_usage$iter__3924__3932$fn__3933$iter__3926__3934$fn__3935.invoke(linters.clj:339)
	at clojure.lang.LazySeq.sval(LazySeq.java:42)
	at clojure.lang.LazySeq.seq(LazySeq.java:51)
	at clojure.lang.RT.seq(RT.java:535)
	at clojure.core$seq__5402.invokeStatic(core.clj:137)
	at clj_kondo.impl.linters$lint_var_usage$iter__3924__3932$fn__3933.invoke(linters.clj:201)
	at clojure.lang.LazySeq.sval(LazySeq.java:42)
	at clojure.lang.LazySeq.seq(LazySeq.java:51)
	at clojure.lang.RT.seq(RT.java:535)
	at clojure.core$seq__5402.invokeStatic(core.clj:137)
	at clojure.core$dorun.invokeStatic(core.clj:3133)
	at clojure.core$doall.invokeStatic(core.clj:3148)
	at clj_kondo.core$run_BANG_.invokeStatic(core.clj:97)
	at clj_kondo.main$main.invokeStatic(main.clj:85)
	at clj_kondo.main$main.doInvoke(main.clj:78)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invokeStatic(core.clj:665)
	at clj_kondo.main$_main$fn__4988.invoke(main.clj:104)
	at clj_kondo.main$_main.invokeStatic(main.clj:104)
	at clj_kondo.main$_main.doInvoke(main.clj:102)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clj_kondo.main.main(Unknown Source)
```

**Comment by borkdude** [comment]
Fixed with https://github.com/borkdude/clj-kondo/commit/ed6754c375bb7a6fadc944de7412d2f9df5321e6.

New Mac binary:

https://7000-176829714-gh.circle-artifacts.com/0/release/clj-kondo-2019.11.04-SNAPSHOT-macos-amd64.zip
